daffodil-core/src/main/scala/org/apache/daffodil/grammar/primitives/PrimitivesDateTime.scala

```diff
@@ -191,3 +186,61 @@ case class ConvertTextDateTimePrim(e: ElementBase) extends ConvertTextCalendarPr
   protected override val implicitPattern = "uuuu-MM-dd'T'HH:mm:ss"
   protected override val validFormatCharacters = "adDeEFGhHkKmMsSuwWvVyXxYzZ".toSeq
 }
+
+abstract class ConvertBinaryCalendarPrimBase(e: ElementBase, guard: Boolean, lengthInBits: Long)
+  extends ConvertCalendarPrimBase(e, guard) {
+
+  lazy val epochCalendar: Calendar = {
```

Comment:
All this stuff in here is specific to binaryCalendarRep="binarySeconds/binaryMilliseconds" so probably shouldn't be in this base class. We'll probably want this base class for the other binaryCalendarReps, so maybe just move it all into ConvertBinaryDateTimePrim? This abstract class will just be empty for now.